### ISSUE TITLE
Error when listing jobs for container that doesn't exist

### ISSUE THREAD

**Issue by coltonlw** [scitran/core#517]
Steps to reproduce:

1) Make a request to list jobs for a container, setting the container ID uri parameter to a non-existent object id

Impact: low

Scope:  Affects all ContainerHandler resources

Example request and traceback:

```
GET /api/sessions/57ebd4c694475a00275358aa/jobs {u'root': u'true'} request_id=9057eda9-1475073225
2016-09-28 14:33:45      scitran.api      containerstorage.py   123:DEBU {'session': ObjectId('57ebd4c694475a00275358aa')}
2016-09-28 14:33:45      scitran.api      containerstorage.py   124:DEBU {'permissions': 0}
2016-09-28 14:33:45      scitran.api                  base.py   286:ERRO Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/webapp2.py", line 570, in dispatch
    return method(*args, **kwargs)
  File "./api/handlers/containerhandler.py", line 232, in get_jobs
    cont = self._get_container(cid, projection={'permissions': 0}, get_children=True)
  File "./api/handlers/containerhandler.py", line 507, in _get_container
    container = self.storage.get_container(_id, projection=projection, get_children=get_children)
  File "./api/dao/containerstorage.py", line 41, in get_container
    cont[child_name] = ContainerStorage(child_name, True).exec_op('GET', query=query, projection=projection)
TypeError: 'NoneType' object does not support item assignment
```

**Comment by ryansanford** [comment]
@coltonlw 
Agree on priority. Added Design label to indicate agreement on consistent behavior when invalid container IDs are requested. I'm going to assign back to you for consideration after RAML spec is complete.

Added

**Comment by nagem** [comment]
The bug itself was fixed via #582.

**Comment by nagem** [comment]
You will still get mildly different error messages when a container is not found, depending on what endpoint/method found the error, but I do not know of any "Not Found" situations that do not return a 404 status code. They might still be there, though 😄.

**Comment by ryansanford** [comment]
Megan will link to this for reference in the Error handing tech debt ticket for future improvements.

**Comment by nagem** [comment]
Fixed in #582
